modules/standard/Path.chpl

```diff
@@ -77,6 +77,26 @@ const parentDir = "..";
 /* Denotes the separator between a directory and its child. */
 const pathSep = "/";
 
+private inline proc _conv(type t, param s) {
+  if t == string then return s;
+  else if t == bytes then return s:bytes;
+  else compilerError("Unrecognized type for conversion");
+}
+
+private inline proc _iconv(type t, val) {
+  if t == string then return val:byteIndex;
+  else if t == bytes then return val:int;
```

Comment:
It doesn't have to be in this PR but I see no reason that the bytes indexing can't accept a byteIndex...